controllers/articles.controller.js

```diff
@@ -1,21 +1,32 @@
-const fetchArticlesById = require("../models/articles.model");
+const {
+  fetchArticleById,
+  fetchArticlesWithCommentCount,
+} = require("../models/articles.model");
 
-const listArticlesById = (req, res, next) => {
+const getArticleById = (req, res, next) => {
   const articleId = req.params.article_id;
-  if(isNaN(articleId)){
+  if (isNaN(articleId)) {
     //convert to number then check if that is actually a number
-    return res.status(400).json({ msg: "Invalid article_id Format. Must Be a Number." });
+    return res
+      .status(400)
+      .json({ msg: "Invalid article_id Format. Must Be a Number." });
   }
-  fetchArticlesById(articleId)
+  fetchArticleById(articleId)
     .then((article) => {
       if (article.length === 0) {
-        res.status(404).json({ msg: "Article Not Found" });
-      } else {
-        res.status(200).json({ article });
-        console.log(`Article by ID:${articleId} >>> `, article);
+        return Promise.reject({ status: 404, msg: "Article Not Found" });
       }
+      res.status(200).json(article[0]);
     })
     .catch(next);
 };
 
-module.exports = listArticlesById;
+const getArticlesByLifo = (req, res, next) => {
+  fetchArticlesWithCommentCount()
+    .then((articles) => {
+      res.json(articles);
+    })
+    .catch(next);
+};
+
```

Comment:
LIFO = Last in first out , i thought that would be a good name since it is desc. order by date. most recent will be given first in the response :)  i ll stick to your recommendation thanks Lee